test/mpi/Makefile_cuda.mtest

```diff
@@ -0,0 +1,26 @@
+##
+## Copyright (C) by Argonne National Laboratory
+##     See COPYRIGHT in top-level directory
+##
+
+## This is an automake makefile fragment that should be included by:
+##
+##   include $(top_srcdir)/Makefile_cuda.mtest
+##
+## This makefile is for CUDA tests (compiled with nvcc).
+
+include $(top_srcdir)/Makefile_single.mtest
+
+AM_DEFAULT_SOURCE_EXT = .cu
+
+AM_V_CCLD = $(am__v_CCLD_$(V))
+am__v_CCLD_ = $(am__v_CCLD_0)
+am__v_CCLD_0 = @echo "  CCLD    " $@;
+am__v_CCLD_1 =
+
+LINK = $(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) \
+       $(LIBTOOLFLAGS) --mode=link $(CXX) $(AM_CFLAGS) $(CFLAGS) \
+       $(AM_LDFLAGS) $(LDFLAGS) -o $@
+
+.cu.o:
+	MPICH_CC=$(NVCC) $(CC) -c -o $@ $<
```

Comment:
Can you add some comment describing what these settings do? I guess the issue is the only source file is a `.cu` file which automake won't know how to link? Not sure how to parse the rule to build objects.

Reply:
Yeah, this is essentially a hack to get `.cu` files working until `automake` and `libtool` support it. Because there isn't a cuda language tag, libtool is not generating them, thus we manually supply them. The `AM_V_` stuff `automake`'s way of support verbose/non-verbose make. The `LINK` is what libtool should generate if it had a cuda language tag. Here we simply copied its `CXX` rule. The `.cu.o` is to compile `.cu` files.

Reply:
I'll add the comments

Reply:
Comments added.